### PR TITLE
EY completed

### PR DESCRIPTION
**Pull request by thisisedyip**

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -11,19 +11,39 @@
...
-
+import random
+from random import randint
 
 # Body
 
-
+def randomguess():
+    guess = ''
+    actual = random.randint(1,25)
```
Comment (danielsgriffin) [inline D04_ex00.py]: If you do `from random import randint` then you only need to do `randint(1, 25)` here.

Comment (danielsgriffin) [inline D04_ex00.py]: please add space: (1, 25)

File: D04_ex00.py [hunk 2]
```diff
@@ -11,19 +11,39 @@
...
+    countdown = 5
+
+    while countdown > 0:
+            try:
+                    guess = float(input("Guess a number: "))
+                    countdown -= 1
+                    if guess > actual and 1<=guess<=25:
```
Comment (danielsgriffin) [inline D04_ex00.py]: Why did you order your condition checks like this? Is `1<=guess<=25` necessary if you switched some around?

Comment (danielsgriffin) [inline D04_ex00.py]: ahh, perhaps because your aren't using elifs and else...

File: HW04_ch07_ex02.py [hunk 3]
```diff
@@ -21,15 +21,27 @@
 
 ###############################################################################
 # Imports
-
+import math
 
 # Body
-
+def eval_loop():
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: "return the value of the last expression it evaluated"
but good job.

File: HW04_ch08_ex05.py [hunk 4]
```diff
@@ -2,3 +2,26 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
 # function.
+
+def rotate_letter(letter, n):
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: Very good.

**Comment by danielsgriffin** [comment]
Good job.

File: HW04_string2.py [hunk 5]
```diff
@@ -17,8 +17,14 @@
...
 def verbing(s):
-    # +++your code here+++
-    return
+    if len(s) >= 3:
+        if s[-3:] != 'ing':
+            newword = s+'ing'
+        elif s[-3:] == 'ing':
+            newword = s+'ly'
+    elif len(s) < 3:
+        newword = s
```
Comment (sheffieldjordan) [inline HW04_string2.py]: I think here you could just `return s`.

File: HW04_string2.py [hunk 6]
```diff
@@ -30,9 +36,20 @@ def verbing(s):
...
+            count += 1           
+        if 'bad' in words and count == 1: #scan for the word 'bad' only if 'not' is found first
+            newword2 = s.split('bad') #print out anything that comes after bad, e.g. punctuation
+            count += 1           
+        if count == 2: #print newly constructed sentence if both not and bad have been found sequentially
+            final =  newword[0] + 'good' + newword2[1]
+        elif count != 2: #print out normal sentence if not and bad were not found sequentially
+            final = s
+            
+    return final
```
Comment (sheffieldjordan) [inline HW04_string2.py]: looks good

**Pull request closed without merging**